File: 1380.cpp
#include <bits/stdc++.h>
using namespace std;

int resto(string n){
    int size = n.size();
    int num = 0,i;
    for(i=0;i<size;i++){
        num = num*10 + ((int)n[i]-48);
        num = num%1500;
    }
    return num%1500;
}

int main(){
    vector<int> lst(1502);
    lst[0] = 0;
    lst[1] = 1;
    for(int i = 2;i<1501;i++){
        lst[i] = ((lst[i-1] + lst[i-2])%1000);
    }
    int inst,num;
    string k;
    cin>>inst;
    for(int i =0;i<inst;i++){
        cin>>k;
        num = resto(k);
        num = lst[num];
        if(num<100){
            if(num<10) cout<<"00";
            else cout<<"0";
        }
        cout<<num<<endl;
    }

}

File: 1380.py
def resto(n):
    size = len(n)
    num = 0
    i = 0
    while(i<size):
        num = num*10 + int(n[i])
        num = num%1500
        i+=1
    return num%1500

lst = [0,1]
for i in range(2,1501):
    lst += [(lst[i-1] + lst[i-2])%1000]
k = int(input())
for i in range(0,k):
    n = input()
    r = resto(n)
    num = lst[r]
    print("{0:03d}".format(num))

